File: workout-pekko/src/main/scala/workout/domain/UpdateWorkoutRequest.scala
package workout.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto._

case class UpdateWorkoutRequest(
    name: Option[String],
    description: Option[String],
    workoutType: Option[WorkoutType],
    durationMinutes: Option[Int],
    caloriesBurned: Option[Int],
    difficulty: Option[Difficulty]
)

object UpdateWorkoutRequest {
  implicit val encoder: Encoder[UpdateWorkoutRequest] = deriveEncoder[UpdateWorkoutRequest]
  implicit val decoder: Decoder[UpdateWorkoutRequest] = deriveDecoder[UpdateWorkoutRequest]
}

File: workout-spring-webflux/src/main/java/com/workout/app/api/dto/UpdateWorkoutRequest.java
package com.workout.app.api.dto;

import com.workout.app.domain.Difficulty;
import com.workout.app.domain.WorkoutType;

public record UpdateWorkoutRequest(
        String name,
        String description,
        WorkoutType workoutType,
        Integer durationMinutes,
        Integer caloriesBurned,
        Difficulty difficulty
) {}

